controller/controller.go

```diff
@@ -262,41 +264,48 @@ func (c *Controller) propagate(resourceEvents []ResourceEvent) {
 }
 
 func (c *Controller) subscribe() {
-	cache, ok := c.cache.(*watchableCacheStore) // should we add Subscribe(ctx) to the Cache interface or remove the interface altogether?
-	if !ok {
-		return
-	}
-	recent := make(Store)
-	subscription := cache.Subscribe(context.TODO())
+	oldObjs := make(Store)
+	subscription := c.cache.SubscribeSubset(context.TODO(), func(storeId string, _ Store) bool {
```

Comment:
I am wondering would it be better to pass a context in the creation of the `Controller` over defining a `context.TODO` here. We already need a context to be passed to the `Controller.Start` function.

Reply:
Sure! Happy to do that. Well spotted. Thanks!

Reply:
Done.